pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -0,0 +1,201 @@
+package gcp
+
+import (
+	"fmt"
+
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// listCloudControllerInstanceGroups returns instance groups created by the cloud controller.
+// It list all instance groups matching the cloud controller name convention.
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_naming.go#L33-L40
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_clusterid.go#L210-L238
+func (o *ClusterUninstaller) listCloudControllerInstanceGroups() ([]cloudResource, error) {
+	filter := fmt.Sprintf("name eq \"k8s-ig--%s\"", o.cloudControllerUID)
+	return o.listInstanceGroupsWithFilter("items/*/instanceGroups(name,zone),nextPageToken", filter, nil)
+}
+
+// listBackendServicesForInstanceGroups returns backend services created by the cloud controller.
+// It list all backend services matching the cloud controller name convention that contain
+// only cluster instance groups.
+func (o *ClusterUninstaller) listBackendServicesForInstanceGroups(instanceGroups []cloudResource, installerInstanceGroups []cloudResource) ([]cloudResource, error) {
```

Comment:
nit: also might be easier to rename this to something like `listCloudControllerTargetPools` `listCloudControllerBackendServices` ?

Reply:
Good idea, done.